### ISSUE TITLE
[ENH] Detect rather than hardcode PAGE_SIZE

### ISSUE THREAD

**Issue by hmacdope** [MDAnalysis/omnireader#2]
On POSIX you can get the page size from a syscall, we should try that and then default to some value, see: https://stackoverflow.com/questions/3351940/detecting-the-memory-page-size

**Comment by richardjgowers** [comment]
Yeah I'm not sure if I literally want a page or if I'm misusing that term.  Ideally we'd benchmark and find a sweet spot.... It might be we want to fit within a cache size or something.  Maybe better would be to expose "pagesize" as an argument and do these tests.  Iirc the only lower bound is that the page has to span two lines.

**Comment by hmacdope** [comment]
Ah okay so more a buffer of set size. Yeah probably good to benchmark, but will be very cache size dependent (ofc).
